src/components/ui-kit/view/ImageViewer/ImageViewer.tsx

```diff
@@ -0,0 +1,268 @@
+    rotation: rotate,
+  });
+
+  const setScale = ({ value, shouldAnimate }: { value: number; shouldAnimate: boolean }) => {
+    lastScale *= value;
+
+    if (shouldAnimate) {
+      Animated.spring(baseScale, {
+        toValue: lastScale,
+        useNativeDriver,
+      }).start();
+      Animated.spring(pinchScale, {
+        toValue: 1,
+        useNativeDriver,
+      }).start();
+    } else {
+      baseScale.setValue(lastScale);
+      pinchScale.setValue(1);
+    }
+  };
+
+  const setTranslate = ({ value, shouldAnimate }: { value: { x: number; y: number }; shouldAnimate: boolean }) => {
+    translateX.setOffset(value.x);
+    translateY.setOffset(value.y);
+    if (shouldAnimate) {
+      Animated.spring(translateX, {
+        toValue: 0,
+        useNativeDriver: true,
+      }).start();
+
+      Animated.spring(translateY, {
+        toValue: 0,
+        useNativeDriver,
+      }).start();
+    } else {
+      translateX.setValue(0);
+      translateY.setValue(0);
+    }
+  };
+
+  const handlePinchStateChange = ({ nativeEvent }: HandlerStateChangeEvent<PinchGestureHandlerEventPayload>) => {
+    if (nativeEvent.oldState === State.ACTIVE) {
+      setScale({
+        value: nativeEvent.scale,
+        shouldAnimate: false,
+      });
+    }
+
+    // Reset the scale and the translate
+    if (nativeEvent.state === State.END) {
+      if (nativeEvent.scale <= 1) {
+        lastScale = 1;
+        setScale({
+          value: lastScale,
+          shouldAnimate: true,
+        });
+
+        lastTranslate = { x: 0, y: 0 };
+        setTranslate({
+          value: {
+            x: 0,
+            y: 0,
+          },
+          shouldAnimate: true,
+        });
+      }
+    }
+  };
+
+  const isImageOutOfBounds = async (): Promise<{ x: boolean; y: boolean }> => {
+    return new Promise((resolve) => {
+      imageRef.current?.measureInWindow((x, y, width, height) => {
+        const { width: windowWidth, height: windowHeight } = Dimensions.get('window');
+        const maxX = width - windowWidth;
+        const maxY = height - windowHeight;
+        const outX = Math.abs(x > 0 ? x + width / 2 : x) > maxX;
+        const outY = Math.abs(y > 0 ? y + height / 2 : y) > maxY;
+        resolve({ x: outX, y: outY });
+      });
+    });
+  };
+  const handleRotationStateChange = ({ nativeEvent }: HandlerStateChangeEvent<RotationGestureHandlerEventPayload>) => {
+    if (nativeEvent.state === State.ACTIVE) {
+      lastRotate += nativeEvent.rotation;
+      rotate.setOffset(lastRotate);
+      rotate.setValue(0);
+    }
+
+    if (nativeEvent.state === State.END) {
+      rotate.setOffset(0);
+      Animated.spring(rotate, {
+        toValue: 0,
+        useNativeDriver,
+      }).start();
+      lastRotate = 0;
+    }
+  };
+
+  const handlePanningStateChange = ({ nativeEvent }: HandlerStateChangeEvent<PanGestureHandlerEventPayload>) => {
+    if (nativeEvent.state === State.END) {
+      isImageOutOfBounds().then(() => {
```

Comment:
WIP, im trying to figure out a way to detect if a photo is moved out of the screen, we don't have CSS magic here.... Ideas are welcome